charts/hivemq-platform/templates/_helpers.tpl

```diff
@@ -475,6 +474,82 @@ Usage: {{- include "hivemq-platform.validate-additional-volumes" . }}
+*/}}
+{{- define "hivemq-platform.validate-licenses" -}}
+{{- $licenseContentExists := "" -}}
+{{- if .Values.license.create -}}
+    {{- include "hivemq-platform.validate-license-content" (dict "licenses" .Values.license.additionalLicenses "licenseLabel" "Additional HiveMQ Broker") -}}
+    {{- include "hivemq-platform.validate-license-content" (dict "licenses" .Values.license.extensions "licenseLabel" "HiveMQ Enterprise Extension") -}}
+    {{- include "hivemq-platform.validate-license-content" (dict "licenses" .Values.license.dataHub "licenseLabel" "HiveMQ Data Hub") -}}
+    {{- if and (not .Values.license.overrideLicense) (not .Values.license.data) (not .Values.license.additionalLicenses) (not .Values.license.extensions) (not .Values.license.dataHub) -}}
+        {{- fail (printf "HiveMQ Platform license content cannot be empty") -}}
+    {{- end -}}
+    {{- if and .Values.license.data .Values.license.overrideLicense -}}
+        {{- fail (printf "Both `data` and `overrideLicense` values are set for the HiveMQ Broker license content. Please, use only one of them") -}}
+    {{- end -}}
+    {{- include "hivemq-platform.validate-duplicated-additional-broker-license" . -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Validates the licenses content within a range function:
+ - At least either `data` or `overrideLicense` is not empty.
+ - Only one of `data` or `overrideLicense` are set, and not both.
+Usage: {{ include "hivemq-platform.validate-license-content" (dict "licenses" .licenses "licenseLabel" "HiveMQ Enterprise Extension") }}
+*/}}
+{{- define "hivemq-platform.validate-license-content" -}}
+{{- $label := .licenseLabel -}}
+{{- range $licenseName, $license := .licenses -}}
+    {{- if and (not $license.data) (not $license.overrideLicense) -}}
+        {{- fail (printf "%s '%s' license content cannot be empty. Please, use either `data` or `overrideLicense` values" $label $licenseName) }}
+    {{- end -}}
+    {{- if and (gt (len $license) 1) (hasKey $license "data") (hasKey $license "overrideLicense") -}}
+        {{- fail (printf "Both `data` and `overrideLicense` values are set for the %s '%s' license content. Please, use only one of them" $label $licenseName) }}
+    {{- else if gt (len $license) 1 -}}
+        {{- fail (printf "Invalid values for setting a license content. Only `data` or `overrideLicense` values are allowed") }}
+    {{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{/*
+Validates there is no "license.lic" defined in the additional broker license values, so it doesn't clash with the default broker name.
+Usage: {{ include "hivemq-platform.validate-duplicated-additional-broker-license" . }}
+*/}}
+{{- define "hivemq-platform.validate-duplicated-additional-broker-license" -}}
+{{- if or .Values.license.data .Values.license.overrideLicense -}}
+    {{- range $licenseName, $license := .Values.license.additionalLicenses -}}
+      {{- $secretEntry := printf "%s%s" $licenseName ".lic" -}}
```

Comment:
Can't we directly check `if eq $licenseName "license"` here?

Reply:
Sorry, having a hard time now with GitHub... I guess you're referring to line 528.
And instead of:
```
{{- if eq $secretEntry "license.lic" -}}
```

Just use:

```
if eq $licenseName "license"
```

If so, yes, we can only check that. No need to include the extension of the file as it will always be `.lic`.

Reply:
Yes, I think the `{{- $secretEntry := printf "%s%s" $licenseName ".lic" -}}` is not needed and the code can be simplified.

Reply:
The current source code is implemented like this, will resolve.